test/readme.test.ts

```diff
@@ -0,0 +1,40 @@
+import { readFileSync } from 'node:fs';
+import path from 'node:path';
+
+import { z } from 'zod';
+
+import { loadConfig } from '../src/config';
+
+/*
+ * Recursively extracts keys from an object.
+ */
+const extractKeys = (item: any): string[] =>
+  typeof item === typeof {}
+    ? [...Object.keys(item), ...Object.values(item).flatMap((element) => extractKeys(element))]
+    : [];
+
+describe('checks README', () => {
+  it('checks that the README contains all configuration keys in airseeker.example.json', () => {
+    const config = loadConfig(path.join(__dirname, '../config'), 'airseeker.example.json');
+    expect(config).toBeDefined();
+
+    const readmeData = readFileSync(path.join(__dirname, '../README.md')).toString();
+    expect(readmeData).toBeDefined();
+
+    // console.log(config);
+    const keys = extractKeys(config).filter((item) => !z.coerce.number().safeParse(item).success);
```

Comment:
You will need some manual filtering for object keys (e.g. hardhat) so might as well just create a the list of properties to filter out (I think only "31337" and "hardhat").